Tests/Tests/CDTQTextSearchTests.m

```diff
@@ -0,0 +1,404 @@
+                          @"age" : @12,
+                          @"pet" : @"cat",
+                          @"comment" : @"He lives in Bristol, UK and his best friend is Mike."};
+            [ds createDocumentFromRevision:rev error:nil];
+            
+            rev.docId = @"john34";
+            rev.body =
+                @{ @"name" : @"john",
+                   @"age" : @34,
+                   @"pet" : @"cat",
+                   @"comment" : @"وهو يعيش في بريستول، المملكة المتحدة، وأفضل صديق له هو مايك."};
+            [ds createDocumentFromRevision:rev error:nil];
+            
+            im = [CDTQIndexManager managerUsingDatastore:ds error:nil];
+            expect(im).toNot.beNil();
+        });
+
+        it(@"can perform a search consisting of a single text clause", ^{
+            expect([im ensureIndexed:@[ @"comment" ]
+                            withName:@"basic_text"
+                                type:@"text"]).toNot.beNil();
+            
+            NSDictionary* query = @{ @"$text" : @{@"$search" : @"lives in Bristol"} };
+            CDTQResultSet* result = [im find:query];
+            expect(result.documentIds).to.containAllElements(@[ @"mike12", @"mike34", @"fred12" ]);
+        });
+        
+        it(@"can perform a phrase search", ^{
+            expect([im ensureIndexed:@[ @"comment" ]
+                            withName:@"basic_text"
+                                type:@"text"]).toNot.beNil();
+            
+            NSDictionary* query = @{ @"$text" : @{@"$search" : @"\"lives in Bristol\""} };
+            CDTQResultSet* result = [im find:query];
+            expect(result.documentIds).to.containAllElements(@[ @"mike12", @"fred12" ]);
+        });
+        
+        it(@"can perform a search containing an apostrophe", ^{
+            expect([im ensureIndexed:@[ @"comment" ]
+                            withName:@"basic_text"
+                                type:@"text"]).toNot.beNil();
+            
+            NSDictionary* query = @{ @"$text" : @{@"$search" : @"he's retired"} };
+            CDTQResultSet* result = [im find:query];
+            expect(result.documentIds).to.containAllElements(@[ @"mike72" ]);
+        });
+        
+        it(@"can perform a search consisting of a single text clause with a sort", ^{
+            expect([im ensureIndexed:@[ @"name", @"comment" ]
+                            withName:@"basic_text"
+                                type:@"text"]).toNot.beNil();
+            
+            NSDictionary* query = @{ @"$text" : @{@"$search" : @"best friend"} };
+            NSArray *order = @[ @{ @"name" : @"asc" } ];
+            CDTQResultSet* result = [im find:query skip:0 limit:0 fields:nil sort: order];
+            expect(result.documentIds).to.containAllElements(@[ @"mike12", @"fred12" ]);
+            expect(result.documentIds[0]).to.equal(@"fred12");
+            expect(result.documentIds[1]).to.equal(@"mike12");
```

Comment:
scratch that, I just noticed it's sorting, sorry, not enough coffee this morning.